src/core/coap/coap.cpp

```diff
@@ -478,22 +539,53 @@ void CoapBase::ProcessReceivedResponse(Message &aMessage, const Ip6::MessageInfo
         if (aMessage.IsEmpty())
         {
             // Empty acknowledgment.
-            if (coapMetadata.mConfirmable)
+#if OPENTHREAD_CONFIG_COAP_OBSERVE_API_ENABLE
+            if (coapMetadata.mObserve && (!request->IsRequest()))
```

Comment:
```suggestion
            if (coapMetadata.mObserve && !request->IsRequest())
```

Reply:
Heh, yeah, I'm paranoid about operator precedence since I bounce between about 6 different languages and the only thing I lean on is the fact that brackets tend to rule the roost.

I'll make a note to tweak that as I re-base.

Reply:
Fix is in 0f64a1612379a0b052cdf08a761ec2a8fe5f099e.